tests/vmi_configuration_test.go

```diff
@@ -566,58 +566,102 @@ var _ = Describe("[sig-compute]Configurations", decorators.SigCompute, func() {
 			})
 		})
 
-		Context("with ACPI SLIC table", func() {
-			It("Should configure guest APCI SLIC with Secret file", func() {
-				const (
-					volumeSlicSecretName = "volume-slic-secret"
-					secretWithSlicName   = "secret-with-slic-data"
-				)
-				var slicTable = []byte{
-					0x53, 0x4c, 0x49, 0x43, 0x24, 0x00, 0x00, 0x00, 0x01, 0x49, 0x43, 0x52,
-					0x41, 0x53, 0x48, 0x20, 0x4d, 0x45, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-					0x88, 0x04, 0x00, 0x00, 0x71, 0x65, 0x6d, 0x75, 0x00, 0x00, 0x00, 0x00,
+		DescribeTable("with ACPI table", func(numTables int, hasSlic, hasMsdm bool) {
+			suffix := func(s string, i int) string {
+				return fmt.Sprintf("%s-%d", s, i)
+			}
+			bytesToHexStr := func(data []byte) string {
+				var s string
+				for _, b := range data {
+					s += fmt.Sprintf("%02x", b)
 				}
-				// To easily compare with console output
-				var hexData string
-				for _, b := range slicTable {
-					hexData += fmt.Sprintf("%02x", b)
+				return s
+			}
+
+			const (
+				volumeSecretName = "volume-secret"
+				secretName       = "secret-with-acpi-data"
+			)
+			var (
+				slicHeader = []byte{0x53, 0x4c, 0x49, 0x43, 0x24, 0x00, 0x00, 0x00, 0x01, 0x49}
+				msdmHeader = []byte{0x4d, 0x53, 0x44, 0x4d, 0x24, 0x00, 0x00, 0x00, 0x01, 0x43}
+				payload    = []byte{
+					0x43, 0x52, 0x41, 0x53, 0x48, 0x20, 0x4d,
+					0x45, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+					0x88, 0x04, 0x00, 0x00, 0x71, 0x65, 0x6d,
+					0x75, 0x00, 0x00, 0x00, 0x00,
 				}
+				slicData = append(slicHeader, payload...)
+				msdmData = append(msdmHeader, payload...)
+				slicSet  = false
+				msdmSet  = false
+			)
 
-				vmi := libvmifact.NewAlpine()
+			vmi := libvmifact.NewAlpine()
+			vmi.Spec.Domain.Firmware = &v1.Firmware{
+				ACPI: &v1.ACPI{},
+			}
 
-				By("Creating a secret with the binary ACPI SLIC table")
-				secret := libsecret.New(secretWithSlicName, libsecret.DataBytes{"slic.bin": slicTable})
-				secret, err := virtClient.CoreV1().Secrets(testsuite.GetTestNamespace(vmi)).Create(context.Background(), secret, metav1.CreateOptions{})
+			for i := range numTables {
```

Comment:
Loops and keeping state in variables is rather hard to keep track of. Can you make it more explicit?

Reply:
I can split the test or try to move things into the Entries() I'll take a look again after the virt rpms are merged

Reply:
I'll move forward and split them. It should be enough to validate that when setting ACPI field and a Secret Volume, we get the data into the Guest firmware table. I mean, I don't tneed to test SLIC & MSDM, one test each is enough to validate both.